src/codelabs/null-safety.md

```diff
@@ -128,38 +132,39 @@ void main() {
 ## Null-aware operators
 
 If a variable or expression is nullable,
-you can use [type promotion](#type-promotion) 
-to access the type's members,
-or you can use null-aware operators to
-handle nullable values.
-
-If based on the flow of the program,
-you know the value of an expression isn't `null`,
-you can use the [null assertion operator](#the-null-assertion-operator-) (`!`),
-but that will throw an exception if the value is `null`.
-
-To handle potentially `null` values you can instead use
-the conditional property access operator (`?.`)
-or null-coalescing operators (`??`)
+you can use [type promotion](#type-promotion)
+to access the type's members.
+You can also use null-aware operators to handle nullable values.
+
+Sometimes the flow of the program tells you that the value of an
+expression cannot be `null`.
+To force Dart to treat that expression as non-nullable,
+add the [null assertion operator](#the-null-assertion-operator-) (`!`).
+If the value does equal `null`, using this operator throws an exception.
+
+To handle potential `null` values, use the conditional property access
+operator (`?.`) or null-coalescing operators (`??`)
 to conditionally access a property or
 provide a default value if `null` respectively.
 
 
 ### Exercise: Conditional property access
 
-If you're unsure that an expression with a nullable type is `null` or not,
-you can use the conditional member access operator (`?.`)
-to conditional execute the remainder of the expression.
+If you don't know that an expression with a nullable type equals `null` or not,
+you can use the conditional member access operator (`?.`).
+With this operator,
+you can execute the remainder of the expression based on the condition that
+`nullableObject` has a value.
 
 <?code-excerpt "null_safety_codelab/bin/null_aware_operators.dart (conditional-property-access)"?>
 ```dart
 // The following calls the 'action' method only if nullableObject is not null
 nullableObject?.action();
 ```
 
-In the code below, try using conditional property access
-in the `stringLength` method to fix the error and
-to return the length of the string or `null` if it is `null`:
+In the following code, try using conditional property access
+in the `stringLength` method. This fixes the error and
+to return the length of the string or `null` if it equals `null`:
```

Comment:
Change "the error and to return the length" to "the error and returns the length" ?